pkgs/tools/inputmethods/ibus-engines/ibus-table/default.nix

```diff
@@ -30,17 +30,32 @@ stdenv.mkDerivation rec {
   '';
 
   buildInputs = [
-    dconf gtk3 gobject-introspection ibus (python3.withPackages (pypkgs: with pypkgs; [ pygobject3 ]))
+    dconf
+    gtk3
+    gobject-introspection
+    ibus
+    (python3.withPackages (pypkgs: with pypkgs; [
+      pygobject3
+      (toPythonModule ibus)
+    ]))
   ];
 
-  nativeBuildInputs = [ autoreconfHook docbook2x pkgconfig python3.pkgs.wrapPython ];
+  nativeBuildInputs = [
+    autoreconfHook
+    docbook2x
+    pkgconfig
+    python3.pkgs.wrapPython
+    wrapGAppsHook
+  ];
 
   postUnpack = ''
     substituteInPlace $sourceRoot/engine/Makefile.am \
       --replace "docbook2man" "docbook2man --sgml"
   '';
 
-  postFixup = "wrapPythonPrograms";
+  postFixup = ''
+    wrapPythonPrograms
```

Comment:
Is this even necessary? I would expect the `python3` `env` from `buildInputs` to be used by `patchShebangs`.

Reply:
All I can say is that @abbradar introduced this line to all ibus input methods in 66b05cd4e68408bf9026bb304f344b87c66feec3.

Reply:
Yeah, and the commit also removed `pygobject3` from deps. `wrapPythonPrograms` is used to put python modules from deps to a wrapper but we already achieve the same with `python3.withPackages` and do not have python modules in deps.

Reply:
IIRC `pygobject3` was removed because it should already have been propagated from `ibus` (I see it was added back after that so propagation apparently broke at some point). `wrapPythonPrograms` was indeed needed to add dependencies to the binary (long before they were propagated from ibus processes because we used `PYTHONPATH` but we moved to `sys.path`). `python3.withPackages` is a different way of handling this so this line is indeed not needed anymore.